golem_messages/validators.py

```diff
@@ -19,3 +19,103 @@ def validate_integer(field_name, value):
             field=field_name,
             value=value,
         )
+
+
+def validate_positive_number_value(value, value_type):
+    return isinstance(value, value_type) and value > 0
+
+
+def validate_positive_integer_value(field_name, value):
+    if not validate_positive_number_value(value, int):
+        raise exceptions.FieldError(
+            "should be a positive integer field",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_not_negative_integer_value(field_name, value):
+    if not isinstance(value, int) or value < 0:
+        raise exceptions.FieldError(
+            "should be a not negative integer field",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_boolean(field_name, value):
+    if not isinstance(value, bool):
+        raise exceptions.FieldError(
+            f"{field_name} should be boolean",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list(field_name, value, min_length, max_length):
+    if not (isinstance(value, list) and
+            (min_length <= len(value) <= max_length)):
+        raise exceptions.FieldError(
+            f"{field_name} should be a list with minimum {min_length}"
+            f" element and maximum {max_length} elements.",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list_of_integers(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not isinstance(element, int) and element > 0:
+            raise exceptions.FieldError(
+                f"{field_name} should be a list with positive integers",
+                field=field_name,
+                value=value
+            )
+
+
+def validate_list_of_floats(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not isinstance(element, float) and element >= 0:
+            raise exceptions.FieldError(
+                f"{field_name} should be a list of floats equal or greater 0",
+                field=field_name,
+                value=element
+            )
+
+
+def validate_enum(field_name, value, enum_class):
+    try:
+        enum_class(value)
+    except ValueError:
+        raise exceptions.FieldError(
+            f"{field_name} has not valid value for"
+            f" {enum_class.__name__} enum class",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_task_type_with_meta_parameters(field_name, value, valid_pairs):
+    for pair in valid_pairs:
+        if pair[0].name == value['task_type']:
+            try:
+                pair[1](value['meta_parameters'])
+            except KeyError:
+                raise exceptions.FieldError(
```

Comment:
This `try ... except` is too broad. Do you expect to catch the `KeyError`:
- from `pair[1]` (when `pair` less than two items)?
- from `value['meta_parameters']` (when `value` does not have that key)?
- from inside the function stored in `pair[1]`?

Only the one from the three you want to cover should be inside the block. The others should be outside because otherwise you'll return a misleading error message.

Reply:
and even looking at the code here, I'm coming towards a conclusion that the usage of the `ValidatingDependentDictItems` is an unnecessary complication ... why use a generalized interface if we later strictly assume the field names? ... it just doesn't seem to make sense to me...